### ISSUE TITLE
Opt: Opaque Inliner

### ISSUE THREAD

**Issue by greg-lunarg** [KhronosGroup/SPIRV-Tools#754]
Create inliner pass which *only* inlines function calls with opaque arguments or return value.

This pass is part of an effort to assist glslang in creating valid SPIR-V from HLSL shaders which store and load opaque objects.

The algorithm will start by performing the inlining only in entry point functions. After the inlining for a function is complete, the functions of the remaining function calls are added to the list to be processed.

Claimed by @greg-lunarg and started.

**Comment by dnovillo** [comment]
@greg-lunarg is this the implementation in opt/inline_opaque_pass.cpp?

**Comment by greg-lunarg** [comment]
Yes. Unfortunately, after writing it, I came to believe it is not useful on its own.

The goal was to create an inliner which would only do the minimum inlining needed to expose all legalization opportunities vis-a-vis opaque types within the entry point function. However, the pass as currently implemented does not do not expose all legalization opportunities.

It could be useful as part of a larger algorithm or it could be expanded to be useful for this purpose.

In the meantime, it was decided that the exhaustive inliner, while not ultimately the desired solution, was sufficient and acceptable for the moment.

So this feature is now on hold.
